mint/mint.go

```diff
@@ -231,6 +249,55 @@ type VerificationParams struct {
 	TargetService string
 }
 
+// Generates a pseudo-random secret.
+func (m *Mint) generatePseudoRandomSecret(id *lsat.Identifier) ([lsat.SecretSize]byte, error) {
+	var ret [lsat.SecretSize]byte
+
+	mac := hmac.New(sha256.New, []byte(*m.cfg.SeedForPseudoRandomness))
+	_, err := mac.Write(id.TokenID[:])
+	if err != nil {
+		return [lsat.SecretSize]byte{}, err
+	}
+
+	result := mac.Sum(nil)
+	if len(result) < lsat.SecretSize {
+		return [lsat.SecretSize]byte{}, ErrIncompatibleHmacFuncUsed
+	}
+
+	copy(ret[0:], result[0:lsat.SecretSize])
+
+	return ret, nil
+}
+
+// Either returns the secret from SecretStore or creates a pseudo-random one.
+func (m *Mint) determineSecret(ctx context.Context, params *VerificationParams, id *lsat.Identifier) ([lsat.SecretSize]byte, error) {
+	if m.cfg.SeedForPseudoRandomness == nil {
+		return m.cfg.Secrets.GetSecret(
+			ctx, sha256.Sum256(params.Macaroon.Id()),
+		)
+	}
+	return m.generatePseudoRandomSecret(id)
+}
+
+// Creates a new secret (persistently stored or pseudo-random).
+func (m *Mint) createSecret(ctx context.Context, secretID [32]byte, id *lsat.Identifier) ([lsat.SecretSize]byte, SecretRemover, error) {
```

Comment:
nit: We try to adhere to `lnd`'s [code guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) and style. For example, everything should be wrapped before column 80 (where possible), with the tab character counting as 8.